tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -203,7 +248,9 @@ def read(self, filename, binary_mode=False, size=None, offset=None):
             if offset is None:
                 offset = 0
             endpoint = '' if size is None else (offset + size)
-            args['Range'] = 'bytes={}-{}'.format(offset, endpoint)
+            if offset != 0 or endpoint != '':
```

Comment:
With our new tests we actually hit a good failure case where the file is empty and we request "0-", which failed in boto. This fixes that issue.